applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/rules.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+'require uci';
+'require fs';
+'require form';
+'require shadowsocks-libev as ss';
+
+var conf = 'shadowsocks-libev';
+
+function src_dst_option(s /*, ... */) {
+	var o = s.taboption.apply(s, L.varargs(arguments, 1));
+	o.datatype = 'or(ipaddr,cidr)';
+}
+
+return L.view.extend({
+	load: function() {
+		return Promise.all([
+			L.resolveDefault(fs.stat('/usr/lib/iptables/libxt_recent.so'), {}),
+			L.resolveDefault(fs.stat('/usr/bin/ss-rules'), null),
```

Comment:
Because `fs.stat` returns a Promise, I moved the binary check in `option_install_package` to here.